packages/db/src/connection.ts

```diff
@@ -49,15 +49,36 @@ export const getKnexClient = async ({
   secretsManager?: AWS.SecretsManager,
   knexLogger?: Logger
 } = {}): Promise<Knex> => {
+  if (isKnexDebugEnabled(env)) {
+    knexLogger.info('Initializing connection pool...');
+  }
   const knexConfig = await getKnexConfig({ env, secretsManager });
   const knexClient = knex(knexConfig);
-  //@ts-ignore
-  // context is an internal object that isn't typed
-  // this is needed to force tarn to log per-retry failures
-  // and allow propagateCreateError to be set `false`
-  knexClient.context.client.pool.on('createFail', (_, error) => {
-    knexLogger.warn('knex failed on attempted connection', error);
-    throw error;
-  });
+  if (isKnexDebugEnabled(env)) {
```

Comment:
This might be overkill, but I think this kind of logging could be very useful for tracing out lambda connection usage. And might be nice to have it enabled easily via an environment variable.